modal/sandbox.py

```diff
@@ -388,6 +389,34 @@ def returncode(self) -> Optional[int]:
         else:
             return self._result.exitcode
 
+    @staticmethod
+    async def list(app_id: Optional[str] = None) -> AsyncGenerator["_Sandbox", None]:
+        """List all sandbox IDs."""
+        before_timestamp = time.time()
+        env = config.get("environment")
+        client = await _Client.from_env()
+        while True:
+            req = api_pb2.SandboxListRequest(
+                app_id=app_id,
+                before_timestamp=before_timestamp,
+                environment_name=env,
+                include_finished=False,
+            )
+
+            # Fetches 100 sandbox IDs at a time.
```

Comment:
100 is hardcoded in the server and I did it this way to avoid having to always take another round trip to confirm there's nothing left. Happy to implement it with that extra round trip and not make any assumptions about the batch size since if the server ever lowered the batch size, the client would never return a complete result. Thoughts?

Reply:
Alternatively you could have the server return a "no more after this" flag or something, maybe that's unclean though.

Reply:
Rewrote without a dependency on batch size